odbc/src/api/data.rs

```diff
@@ -1183,7 +1209,7 @@ pub mod i32_len {
             text_length_ptr,
             (size_of::<WideChar>() * len)
                 .try_into()
-                .unwrap_or(Integer::MAX),
+                .expect("Data too large to fit in the buffer on this platform"),
```

Comment:
It's actually not quite true. The data is too large for the type text_length_ptr points to.

```suggestion
                .expect("Data too large to fit"),
```

If you look at the ODBC API, text_length_ptr and buffer_len are the same type.
We should do the same. This would make things consistent and if a message is too large, then a truncation warning will have to be reported by `set_output_wstring_helper`.